products/libllbuild/BuildDB-C-API.cpp

```diff
@@ -111,6 +115,50 @@ namespace {
 
 }
 
+const llb_data_t mapData(std::vector<uint8_t> input) {
+  const auto data = input.data();
+  const auto size = sizeof(data) * input.size();
+  const auto newData = (uint8_t *)malloc(size);
+  memcpy(newData, data, size);
+  
+  return llb_data_t {
+    input.size(),
+    newData
+  };
+}
+
+const llb_database_result_t mapResult(CAPIBuildDB &db, Result result) {
+  
+  auto count = result.dependencies.size();
+  std::vector<llb_build_key_t> buildKeys;
```

Comment:
Bearing in mind I haven't thought about this deeply, but is there any way to avoid the double multiple allocations that are happening here.   i.e. can we just calculate how much we'll need and splat it into a malloc'd buffer directly?

Reply:
Yes, that's possible 👍